### ISSUE TITLE
Deprecation notice when using raiseThrowables

### ISSUE THREAD

**Issue by martinezdelariva** [zendframework/zend-expressive#431]
It seems like `raiseThrowables` is not well propagated due executes `Next::123` instead of `Next::119`:

```php
// Next.php
class Next implements DelegateInterface
{
    public function __invoke(
        ServerRequestInterface $request,
        ResponseInterface $response,
        $err = null
    ) {
        if ($err !== null && $this->raiseThrowables) {
            $this->raiseThrowableFromError($err);
        }

        if (null !== $err) {
            $this->triggerErrorDeprecation();
        }
     // ...
    }
// ...
}
```

I suspect that is at `MarshalMiddlewareTrait::94` but I'm not pretty sure.

Example
`GET /foo`

```
// expected
HTTP/1.1 500 Internal Server Error

Custom exception
```

```
// actual
HTTP/1.1 500 Internal Server Error

Usage of error middleware is deprecated as of 1.3.0, and will be removed in 2.0.0; please see https://docs.zendframework.com/zend-stratigility/migration/to-v2/ for details on how to update your application to remove this message.
```

Setup:
* zendframework/zend-stratigility 1.3.3
* zendframework/zend-expressive 1.0.6

```php
// index.php

$app = $container->get(\Zend\Expressive\Application::class);
$app->raiseThrowables();

$app->get('/foo', function ($request, $response, \Zend\Stratigility\Next $next = null) {
   throw new Exception("Custom exception"); 
});
```

```php
// middleware-pipeline.global.php

use Zend\Expressive\Container\ApplicationFactory;
use Zend\Expressive\Helper;

return [
    'dependencies' => [
        'factories' => [
            Helper\ServerUrlMiddleware::class => Helper\ServerUrlMiddlewareFactory::class,
            Helper\UrlHelperMiddleware::class => Helper\UrlHelperMiddlewareFactory::class,
        ],
    ],
    'middleware_pipeline' => [
        'always' => [
            'middleware' => [
                Zend\Stratigility\Middleware\ErrorHandler::class,
                Helper\ServerUrlMiddleware::class,
            ],
            'priority' => 10000,
        ],

        'routing' => [
            'middleware' => [
                ApplicationFactory::ROUTING_MIDDLEWARE,
                ApplicationFactory::DISPATCH_MIDDLEWARE,
                Zend\Stratigility\Middleware\NotFoundHandler::class,
            ],
            'priority' => 1,
        ],
    ],
];
```

```php
use Zend\Expressive\Application;
use Zend\Expressive\Container\ApplicationFactory;
use Zend\Expressive\Helper;

return [
    'dependencies' => [
        'invokables' => [
            Helper\ServerUrlHelper::class  => Helper\ServerUrlHelper::class,
            'Zend\Expressive\FinalHandler' => \Zend\Stratigility\NoopFinalHandler::class,
        ],
        'factories' => [
            Application::class => ApplicationFactory::class,
            Helper\UrlHelper::class => Helper\UrlHelperFactory::class,
            \Zend\Stratigility\Middleware\NotFoundHandler::class => function () {
                return new \Zend\Stratigility\Middleware\NotFoundHandler(new \Zend\Diactoros\Response());
            },
            \Zend\Stratigility\Middleware\ErrorHandler::class => function () {
                return new \Zend\Stratigility\Middleware\ErrorHandler(
                    new \Zend\Diactoros\Response(),
                    function ($e, $request, $responsePrototype) {
                        $responsePrototype = $responsePrototype->withStatus(500);
                        $responsePrototype->getBody()->write($e->getMessage());

                        return $responsePrototype;
                    }
                );
            }
        ],
    ],
];
```

**Comment by weierophinney** [comment]
Indeed, the `MarshalMiddlewareTrait` does not call `raiseThrowables()` on `MiddlewarePipe` instances it creates. This is due to the fact that the minimum supported Stratigility version does not define that method, making it unsafe to call currently.

We're planning a 1.1.0 version that will define several deprecations based on the proposed 2.0 API; I'll add this to the 1.1.0 milestone so that we can (a) bump the Stratigility minimum version to a 1.3 series release, and (b) ensure that the flag is propagated to `MiddlewarePipe` instances we create.

**Comment by weierophinney** [comment]
#432 will fix this for the Expressive 1 series.

**Comment by weierophinney** [comment]
Closed with #432.
